internal/controller/repository/repository.go

```diff
@@ -977,6 +1006,448 @@ func handleBranchProtectionSignature(ctx context.Context, gh *ghclient.Client, o
+		// This prevents the controller from changing the spec of the live CR
+		// It can also prevent infinite reconciliation loops when managing the resources with ArgoCD
+		orig := &rules[i]
+		rCopy := orig.DeepCopy()
+
+		// handle optional fields
+		rCopy.Target = util.StringDerefToPointer(rCopy.Target, "branch")
+		rCopy.Enforcement = util.StringDerefToPointer(rCopy.Enforcement, "active")
+
+		rConditions := rCopy.Conditions
+
+		if rConditions != nil && rConditions.RefName != nil {
+			if rConditions.RefName.Include != nil {
+				rConditions.RefName.Include = util.SortAndReturn(rConditions.RefName.Include)
+			}
+			if rConditions.RefName.Exclude != nil {
+				rConditions.RefName.Exclude = util.SortAndReturn(rConditions.RefName.Exclude)
+			}
+		}
+
+		if rConditions == nil {
+			rConditions = &v1alpha1.RulesetConditions{
+				RefName: &v1alpha1.RulesetRefName{
+					Include: []string{},
+					Exclude: []string{},
+				},
+			}
+			// Update the rConditions reference in rCopy
+			rCopy.Conditions = rConditions
+		}
+
+		rBActors := rCopy.BypassActors
+		if rBActors != nil {
+			for a := range rBActors {
+				actor := rBActors[a] // Make a copy of the actor
+
+				// Set ActorId, ActorType, and BypassMode fields
+				actor.ActorId = rBActors[a].ActorId
+				actor.ActorType = rBActors[a].ActorType
+				actor.BypassMode = rBActors[a].BypassMode
+
+				// Update the actor in the slice
+				rBActors[a] = actor
+			}
+			util.SortRulesBypassActors(rBActors)
+		}
+		rRules := rCopy.Rules
+		if rRules != nil {
+			rRules.RequiredSignatures = util.BoolDerefToPointer(rRules.RequiredSignatures, false)
+			rRules.NonFastForward = util.BoolDerefToPointer(rRules.NonFastForward, false)
+			rRules.Creation = util.BoolDerefToPointer(rRules.Creation, false)
+			rRules.Deletion = util.BoolDerefToPointer(rRules.Deletion, false)
+			rRules.RequiredLinearHistory = util.BoolDerefToPointer(rRules.RequiredLinearHistory, false)
+			rRules.Update = util.BoolDerefToPointer(rRules.Update, false)
+
+			if rRules.RequiredDeployments != nil {
+				if rRules.RequiredDeployments.Environments != nil {
+					rRules.RequiredDeployments.Environments = util.SortAndReturn(rRules.RequiredDeployments.Environments)
+				}
+			}
+			if rRules.PullRequest != nil {
+				rRules.PullRequest.DismissStaleReviewsOnPush = util.BoolDerefToPointer(rRules.PullRequest.DismissStaleReviewsOnPush, false)
+				rRules.PullRequest.RequireCodeOwnerReview = util.BoolDerefToPointer(rRules.PullRequest.RequireCodeOwnerReview, false)
+				rRules.PullRequest.RequireLastPushApproval = util.BoolDerefToPointer(rRules.PullRequest.RequireLastPushApproval, false)
+				rRules.PullRequest.RequiredReviewThreadResolution = util.BoolDerefToPointer(rRules.PullRequest.RequiredReviewThreadResolution, false)
+				rRules.PullRequest.RequiredApprovingReviewCount = util.IntDerefToPointer(rRules.PullRequest.RequiredApprovingReviewCount, 0)
+			}
+			//if rRules.RequiredStatusChecks == nil {
```

Comment:
What's up with this commented out code?

Reply:
i was testing status checks functionality but that commented stuff is not needed anymore, forgot to remove